src/generator.ts

```diff
@@ -1014,15 +1011,26 @@ export class Generator {
   async install(
     install?: string | Install | (string | Install)[]
   ): Promise<void | { staticDeps: string[]; dynamicDeps: string[] }> {
-    if (arguments.length > 1)
-      throw new JspmError(
-        "Install takes no arguments, a single install target, or a list of install targets."
-      );
+    return this._install(install);
+  }
+
+  private async _install(
+    install?: string | Install | (string | Install)[],
+    mode?: InstallMode
+  ): Promise<void | { staticDeps: string[]; dynamicDeps: string[] }> {
+    // Backwards-compatibility for deprecated options:
+    if (this.latest) mode ??= "latest-primaries";
+    if (this.freeze) mode ??= "freeze";
 
     // If there are no arguments, then we reinstall all the top-level locks:
-    if (!install) {
+    if (install === null || install === undefined) {
```

Comment:
This was previously handled via `generator.reinstall`. I believe this change means that `reinstall` is now deprecated?

Reply:
The `reinstall()` method is equivalent to `freeze` installing all the top-level locks (or equivalently doing an argumentless `link`, since `link` uses freeze by default). So I think it can be deprecated now in favour of either of those, yeah.

Reply:
Ok, let's do that then, either with this PR or via a follow-up after it lands.